frameworks/mljarsupervised/exec.py

```diff
@@ -15,49 +16,60 @@
 def run(dataset, config):
     log.info("\n**** mljar-supervised ****\n")
 
-    column_names, _ = zip(*dataset.columns)
-    column_types = dict(dataset.columns)
-    X_train = pd.DataFrame(dataset.train.X, columns=column_names).astype(
-        column_types, copy=False
-    )
-    X_test = pd.DataFrame(dataset.test.X, columns=column_names).astype(
-        column_types, copy=False
+    # Mapping of benchmark metrics to MLJAR metrics
+    metrics_mapping = dict(
+        auc='auc',
+        logloss='logloss',
+        rmse='rmse'
     )
-
-    y_train = dataset.train.y.flatten()
-    y_test = dataset.test.y.flatten()
-
+    eval_metric = metrics_mapping[config.metric] if config.metric in metrics_mapping else "auto"
+    
+    # Mapping of benchmark task to MLJAR ML task
     problem_mapping = dict(
         binary="binary_classification",
         multiclass="multiclass_classification",
         regression="regression",
     )
-    is_classification = config.type == "classification"
     ml_task = problem_mapping.get(
         dataset.problem_type
     )  # if None the AutoML will guess about the ML task
+    is_classification = config.type == "classification"
     results_path = output_subdir("results", config)
     training_params = {
         k: v for k, v in config.framework_params.items() if not k.startswith("_")
     }
 
+    column_names, _ = zip(*dataset.columns)
+    column_types = dict(dataset.columns)
+    label = dataset.target.name
+
+    train = pd.DataFrame(dataset.train.data, columns=column_names).astype(column_types, copy=False)
+    X_train = train.drop(columns=label)
+    y_train = train[label]
+
+    test = pd.DataFrame(dataset.test.data, columns=column_names).astype(column_types, copy=False)
+    X_test = test.drop(columns=label)
+    y_test = test[label]
+
     automl = AutoML(
         results_path=results_path,
         total_time_limit=config.max_runtime_seconds,
-        seed=config.seed,
+        random_state=config.seed,
         ml_task=ml_task,
+        eval_metric=eval_metric,
         **training_params
     )
 
     with utils.Timer() as training:
         automl.fit(X_train, y_train)
 
-    preds = automl.predict(X_test)
+    preds = automl.predict_all(X_test)
 
     predictions, probabilities = None, None
     if is_classification:
         predictions = preds["label"].values
-        probabilities = preds[preds.columns[:-1]].values
+        cols = [f"prediction_{c}" for c in np.unique(y_train)]
```

Comment:
Yes, I have column names in `preds`. My columns are in the format: `prediction_{class_name}` and it looks like that if I will use `probabilities_labels` from my `preds` data frame then it will be some mismatch. Looks like `results.py` requires columns to be exactly the same as class values. To not do column renaming I just reorder the columns and pass as array. I tested it, and it works this way.

Reply:
>  Looks like results.py requires columns to be exactly the same as class values.

correct, this is used to ensure consistent column order.

> To not do column renaming I just reorder the columns and pass as array. I tested it, and it works this way.

good, I leave it on you then.